packages/core/e2e/customer-channel.e2e-spec.ts

```diff
@@ -0,0 +1,487 @@
+                input: {
+                    emailAddress: 'john.doe.2@test.com',
+                },
+            });
+
+            adminClient.setChannelToken(SECOND_CHANNEL_TOKEN);
+            const { customers } = await adminClient.query<GetCustomerList.Query, GetCustomerList.Variables>(
+                GET_CUSTOMER_LIST,
+            );
+            expect(customers.totalItems).toBe(4);
+            expect(customers.items.map(customer => customer.emailAddress)).toContain('john.doe.2@test.com');
+        });
+    });
+
+    describe('Customergroup manipulation', () => {
+        it('does not add a customer from another channel to customerGroup', async () => {
+            adminClient.setChannelToken(SECOND_CHANNEL_TOKEN);
+            await adminClient.query<AddCustomersToGroup.Mutation, AddCustomersToGroup.Variables>(
+                ADD_CUSTOMERS_TO_GROUP,
+                {
+                    groupId: customerGroupId,
+                    customerIds: [thirdCustomer.id],
+                },
+            );
+
+            adminClient.setChannelToken(E2E_DEFAULT_CHANNEL_TOKEN);
+            const { customerGroup } = await adminClient.query<
+                GetCustomerGroup.Query,
+                GetCustomerGroup.Variables
+            >(GET_CUSTOMER_GROUP, {
+                id: customerGroupId,
+            });
+            expect(customerGroup!.customers.totalItems).toBe(0);
+        });
+
+        it('only shows customers from current channel in customerGroup', async () => {
+            adminClient.setChannelToken(E2E_DEFAULT_CHANNEL_TOKEN);
+            await adminClient.query<AddCustomersToGroup.Mutation, AddCustomersToGroup.Variables>(
+                ADD_CUSTOMERS_TO_GROUP,
+                {
+                    groupId: customerGroupId,
+                    customerIds: [secondCustomer.id, thirdCustomer.id],
+                },
+            );
+
+            adminClient.setChannelToken(SECOND_CHANNEL_TOKEN);
+            const { customerGroup } = await adminClient.query<
+                GetCustomerGroup.Query,
+                GetCustomerGroup.Variables
+            >(GET_CUSTOMER_GROUP, {
+                id: customerGroupId,
+            });
+            expect(customerGroup!.customers.totalItems).toBe(1);
+            expect(customerGroup!.customers.items.map(customer => customer.id)).toContain(secondCustomer.id);
+        });
+
+        it('throws when deleting customer from other channel from customerGroup', async () => {
+            adminClient.setChannelToken(SECOND_CHANNEL_TOKEN);
+            await adminClient.query<RemoveCustomersFromGroup.Mutation, RemoveCustomersFromGroup.Variables>(
+                REMOVE_CUSTOMERS_FROM_GROUP,
+                {
+                    groupId: customerGroupId,
+                    customerIds: [thirdCustomer.id],
+                },
+            );
+        });
+    });
+});
+
+export const CREATE_ADDRESS = gql`
```

Comment:
These gql operations seem to be identical to the ones in the customer.e2e-spec.ts file.
In that case, my suggestion is to move them into /graphql/shared-definition.ts and import from there into both spec files to avoud duplication.

Reply:
That is how it was originally coded, but there seems to be a bug (feature?) happening here where if you import from another file that also contains tests, Jest starts executing those tests as well. This creates interference issues and the tests actually fail. A separate util file could be created that contains these definitions so they can be shared?

Reply:
I suggest creating an `admin-definitions.ts` file in `core/e2e/graphql` so that the queries are not spread across the tests. The `E2E_ADMIN_QUERY_FILES` constant in `generate-graphql-types.ts` can then be simplified as well. This deserves its own PR in my opinion.

Reply:
The existing `shared-definitions.ts` file already does this, but only with definitions used in more than 1 spec file. I'm not sure I agree with putting absolutely _all_ definitions into a single file because of the resulting size of the file and potential increased difficulty finding what you are looking for. But you can PR it and then we see how it looks, if you like.